storage/backend_key.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"log"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// backendKeyBytesLen is the length of backendKey bytes.
+// First a few bytes stores the revision of backendKey. The following byte
+// is one letter for action: 'p' for put, 't' for tombstone.
+const backendKeyBytesLen = revBytesLen + 1
+
+const (
+	actionPut       action = 'p'
+	actionTombstone action = 't'
+)
+
+type action byte
+
+func (a action) eventType() storagepb.Event_EventType {
```

Comment:
as we discussed here: https://github.com/coreos/etcd/pull/3842#discussion_r44368066, I do not think we need the eventType.